### DISCUSSION TITLE
Inconsistent handling of String columns

### DISCUSSION THREAD

**Discussion by Jammjammjamm** [jeremyevans/sequel#1811] [General]
According to [the documentation](http://sequel.jeremyevans.net/rdoc/files/doc/schema_modification_rdoc.html#label-Column+types), a column type of `String` corresponds to `varchar(255)`. I have a migration which looks like this:
```
    create_table :headers do
      ...
      column :type, String
      column :name, String
      column :value, String
    end
```
When using sqlite, I get a schema like I would expect:
```
    create_table(:headers, :ignore_index_errors=>true) do
      ...
      String :type, :size=>255
      String :name, :size=>255
      String :value, :size=>255
    end
```
But when using postgres, these columns are converted to text:
```
    create_table(:headers, :ignore_index_errors=>true) do
      ...
      String :type, :text=>true
      String :name, :text=>true
      String :value, :text=>true
    end
```

**Comment by adam12** [comment]
> But when using postgres, these columns are converted to text:

This is expected and documented on that page:

> So while you specified String, Sequel will actually use varchar or text depending on the underlying database.

**Comment by Jammjammjamm** [comment]
I that case, I think it would be very helpful to also indicate that uncertainty in the code snippets in that section of the documentation, e.g.
```
String :a1                        # varchar(255)
```
-->
```
String :a1                        # varchar(255)/text (database dependent)
```
I can make a PR if that would be welcome.

**Comment by jeremyevans** [comment]
The heading of the section states `# common database type used`, and `varchar(255)` is the most common. If we were listing all possible specific database types that could be used for each generic type, that would result in quite cluttered examples.
